otelconf/config_test.go

```diff
@@ -10,6 +10,246 @@ import (
+			wantErrT:   errUnmarshalingBatchSpanProcessor,
+		},
+		{
+			name:       "invalid export_timeout negative",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"export_timeout":-1}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nexport_timeout: -1"),
+			wantErrT:   newErrGreaterOrEqualZero("export_timeout"),
+		},
+		{
+			name:       "invalid max_export_batch_size zero",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"max_export_batch_size":0}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nmax_export_batch_size: 0"),
+			wantErrT:   newErrGreaterThanZero("max_export_batch_size"),
+		},
+		{
+			name:       "invalid max_export_batch_size negative",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"max_export_batch_size":-1}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nmax_export_batch_size: -1"),
+			wantErrT:   newErrGreaterThanZero("max_export_batch_size"),
+		},
+		{
+			name:       "invalid max_queue_size zero",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"max_queue_size":0}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nmax_queue_size: 0"),
+			wantErrT:   newErrGreaterThanZero("max_queue_size"),
+		},
+		{
+			name:       "invalid max_queue_size negative",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"max_queue_size":-1}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nmax_queue_size: -1"),
+			wantErrT:   newErrGreaterThanZero("max_queue_size"),
+		},
+		{
+			name:       "invalid schedule_delay negative",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"schedule_delay":-1}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nschedule_delay: -1"),
+			wantErrT:   newErrGreaterOrEqualZero("schedule_delay"),
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			cl := BatchSpanProcessor{}
+			err := cl.UnmarshalJSON(tt.jsonConfig)
+			assert.ErrorIs(t, err, tt.wantErrT)
+
+			cl = BatchSpanProcessor{}
+			err = yaml.Unmarshal(tt.yamlConfig, &cl)
+			assert.ErrorIs(t, err, tt.wantErrT)
+		})
+	}
+}
+
+func TestUnmarshalPeriodicMetricReader(t *testing.T) {
+	for _, tt := range []struct {
+		name       string
+		yamlConfig []byte
+		jsonConfig []byte
+		wantErrT   error
+	}{
+		{
+			name:       "valid with console exporter",
+			jsonConfig: []byte(`{"exporter":{"console":{}}}`),
+			yamlConfig: []byte("exporter:\n  console: {}"),
+		},
+		{
+			name:       "valid with all fields positive",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"timeout":5000,"interval":1000}`),
+			yamlConfig: []byte("exporter:\n  console: {}\ntimeout: 5000\ninterval: 1000"),
+		},
+		{
+			name:       "valid with zero export_timeout",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"export_timeout":0}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nexport_timeout: 0"),
+		},
+		{
+			name:       "valid with zero schedule_delay",
+			jsonConfig: []byte(`{"exporter":{"console":{}},"schedule_delay":0}`),
+			yamlConfig: []byte("exporter:\n  console: {}\nschedule_delay: 0"),
```

Comment:
The `PeriodicMetricReader` doesn't have `export_timeout` or `schedule_delay` fields. It has `timeout` and `interval` fields instead.

```suggestion
			jsonConfig: []byte(`{"exporter":{"console":{}},"timeout":0}`),
			yamlConfig: []byte("exporter:\n  console: {}\ntimeout: 0"),
		},
		{
			name:       "valid with zero schedule_delay",
			jsonConfig: []byte(`{"exporter":{"console":{}},"interval":0}`),
			yamlConfig: []byte("exporter:\n  console: {}\ninterval: 0"),
```

Or, is this a test for the default value of `0` being interpreted when unset?